src/core/thread/address_resolver.cpp

```diff
@@ -162,6 +162,40 @@ void AddressResolver::InvalidateCacheEntry(Cache &aEntry)
     otLogInfoArp(GetInstance(), "cache entry removed!");
 }
 
+void AddressResolver::UpdateCacheEntry(const Ip6::Address &aEid, Mac::ShortAddress aRloc16)
+{
+    for (int i = 0; i < kCacheEntries; i++)
+    {
+        if (mCache[i].mState == Cache::kStateInvalid ||
+            memcmp(&mCache[i].mTarget, &aEid, sizeof(mCache[i].mTarget)) != 0)
+        {
+            continue;
+        }
+
+        if (mCache[i].mRloc16 != aRloc16)
+        {
+            // not updating the age here is intentional because this cache entry is not actually being used
+            mCache[i].mRloc16 = aRloc16;
```

Comment:
What about the case where the entry may be used from `HandleResolved()` call below?

Reply:
Can you provide more detail on your concern?  `HandleResolved()` is only called below when the entry state is `kStateQuery`.  This code block will put the entry into `kStateCached`.

Reply:
Talked to @jwhui  offline. 
If state changes, age will be updated from follow-up `Resolve()` call when a message is being sent.